hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -290,7 +291,7 @@ public SecurityContext getSecurityContext() {
 
     @Override
     public boolean bind(final ClientEndpoint endpoint) {
-        if (!clientSelector.select(endpoint)) {
+        if (isMCClient(endpoint) && !clientSelector.select(endpoint)) {
```

Comment:
It may clearer if you do if MCClient check within the `clientSelector.select` method

Reply:
Thanks for the suggestion. It appeared that I can simply use `#isClientAllowed` method in all of such places.